### PR TITLE
Babylon vienna congress

### PR DESCRIPTION
**Pull request by HiddeLekanne**

### CODE REVIEW

File: common/history/global/babylon_global.txt [hunk 1]
```diff
@@ -37,7 +37,9 @@
             }
         }
     }
-
+	### STRATEGIC REGIONS LIST FOR VIENNA CONGRESS ###
+	make_strategic_region_list = yes
+
```
Comment (HiddeLekanne) [inline common/history/global/babylon_global.txt:42]: weird indentation

File: common/modifiers/babylon_modifiers.txt [hunk 2]
```diff
@@ -444,7 +444,45 @@ babylon_state_infamy = {
...
+
+babylon_vienna_congress_scaling_positive_prestige_modifier = {
+	icon = gfx/interface/icons/timed_modifier_icons/modifier_flag_positive.dds
+	country_prestige_mult = 0.01
+}
+
 babylon_greener_grass_modifier = {
 	icon = gfx/interface/icons/decree/decree_greener_grass_campaign.dds
 	state_migration_pull_mult = 1
 }
```
Comment (HiddeLekanne) [inline common/modifiers/babylon_modifiers.txt:469]: why is this here?

File: common/on_actions/babylon_on_monthtly_pulse.txt [hunk 3]
```diff
@@ -118,6 +128,13 @@ on_babylon_monthly_pulse_store_infamy = {
 			name = infamy_on_month_start
 			value = infamy
 		}
+		if = {
+			limit = {
+				# This should only pop for a singular country each month. Most likely a united China
+				has_global_highest_gdp = yes
+			}
+			babylon_vienna_congress_compile_lists = yes
```
Comment (HiddeLekanne) [inline common/on_actions/babylon_on_monthtly_pulse.txt:136]: just to random_country, it's more efficient and more sensible.

File: common/on_actions/babylon_on_monthtly_pulse.txt [hunk 4]
```diff
@@ -164,3 +181,145 @@ on_babylon_monthly_pulse_state = {
...
+				}
+				# Weights have been divided into 3 pieces
+				# --- : -35
+				# -35 : 35
+				# 35 : ---
+				# Dead zones are between +/-45 and +/-25, they will stay on their chosen stance, prevents flip-flopping
+				if = {
+					# If the voting weight is between above 45 (and you're not supporting)
+					limit = {
+						babylon_discord_ai_voting_weight > 45
```
Comment (HiddeLekanne) [inline common/on_actions/babylon_on_monthtly_pulse.txt]: `>=`

File: common/on_actions/babylon_on_monthtly_pulse.txt [hunk 5]
```diff
@@ -164,3 +181,145 @@ on_babylon_monthly_pulse_state = {
...
+					# Set to stance 1: Supporters
+					set_congress_stance_effect = {
+						COUNTRY = this
+						STANCE = 1
+					}
+				}
+				else_if = {
+					# Else if the voting weight is below -45 (and you're not opposing)
+					limit = {
+						babylon_discord_ai_voting_weight < -45
```
Comment (HiddeLekanne) [inline common/on_actions/babylon_on_monthtly_pulse.txt]: `<=`

File: common/on_actions/babylon_on_monthtly_pulse.txt [hunk 6]
```diff
@@ -164,3 +181,145 @@ on_babylon_monthly_pulse_state = {
...
+				else_if = {
+					# Else if the voting weight is in the middle (and you're not abstaining)
+					limit = {
+						OR = {
+							AND = {
+								is_target_in_global_variable_list = {
+									name = current_congress_members_on_stance_1
+									target = this
+								}
+								babylon_discord_ai_voting_weight < 25
```
Comment (HiddeLekanne) [inline common/on_actions/babylon_on_monthtly_pulse.txt]: `<=`

File: common/on_actions/babylon_on_monthtly_pulse.txt [hunk 7]
```diff
@@ -164,3 +181,145 @@ on_babylon_monthly_pulse_state = {
...
+							}
+							AND = {
+								babylon_discord_ai_voting_weight < 25
+								babylon_discord_ai_voting_weight > -25
```
Comment (HiddeLekanne) [inline common/on_actions/babylon_on_monthtly_pulse.txt]: `>=`

File: common/on_actions/babylon_on_monthtly_pulse.txt [hunk 8]
```diff
@@ -164,3 +181,145 @@ on_babylon_monthly_pulse_state = {
...
+							}
+							AND = {
+								is_target_in_global_variable_list = {
+									name = current_congress_members_on_stance_3
+									target = this
+								}
+								babylon_discord_ai_voting_weight > -25
```
Comment (HiddeLekanne) [inline common/on_actions/babylon_on_monthtly_pulse.txt]: `>=`

File: common/on_actions/babylon_on_monthtly_pulse.txt [hunk 9]
```diff
@@ -164,3 +181,145 @@ on_babylon_monthly_pulse_state = {
...
+						OR = {
+							AND = {
+								is_target_in_global_variable_list = {
+									name = current_congress_members_on_stance_1
+									target = this
+								}
+								babylon_discord_ai_voting_weight < 25
+							}
+							AND = {
+								babylon_discord_ai_voting_weight < 25
```
Comment (HiddeLekanne) [inline common/on_actions/babylon_on_monthtly_pulse.txt]: `<=`

File: common/script_values/babylon_values.txt [hunk 10]
```diff
@@ -748,3 +748,435 @@ babylon_player_construction_bonus_multiplier = {
...
+	value = conference_opposers_prestige_value
+	divide = total_prestige_of_voting_conference_members
+}
+
+conference_power_balance_percentage_value = {
+	value = conference_supporters_percentage_value
+	subtract = conference_opposers_percentage_value
+}
+
+conference_sanctions_severity = {
```
Comment (HiddeLekanne) [inline common/script_values/babylon_values.txt:914]: These severities should be script values at this point.

File: common/scripted_effects/babylon_vienna_congress_effects.txt
Comment (HiddeLekanne) [inline common/scripted_effects/babylon_vienna_congress_effects.txt:1]: Just needs to be tested, not going to review this.

File: events/conference_events/babylon_vienna_congress_events.txt [hunk 11]
```diff
@@ -0,0 +1,215 @@
+﻿namespace = babylon_vienna_congress
+# babylon_vienna_congress.1 = {
```
Comment (HiddeLekanne) [inline events/conference_events/babylon_vienna_congress_events.txt]: Just delete this if it's all commented out anyway

File: events/conference_events/babylon_vienna_congress_random_events.txt
Comment (HiddeLekanne) [inline events/conference_events/babylon_vienna_congress_random_events.txt:1]: Not going to review this, just needs testing

File: common/modifiers/babylon_modifiers.txt (hunk 2 above)
Comment (Danarca) [reply common/modifiers/babylon_modifiers.txt:469]: Git blame says its Fallonars addition 3 weeks ago, so possibly something outdated from Core?

File: common/on_actions/babylon_on_monthtly_pulse.txt (hunk 3 above)
Comment (Danarca) [reply common/on_actions/babylon_on_monthtly_pulse.txt:136]: Yea but this is already in a random_country. If I do random_country again this would just pop 300+ times on random targets =/

File: common/script_values/babylon_values.txt (hunk 10 above)
Comment (Danarca) [reply common/script_values/babylon_values.txt:914]: We use the severity in multipliers, so we still need the number =/ 
So I'm a bit unsure of what you're asking for here

File: common/history/global/babylon_global.txt (hunk 1 above)
Comment (Danarca) [reply common/history/global/babylon_global.txt:42]: It's in `GLOBAL = { ` so its supposed to be indented. Although I agree it kinda looks like 8 spaces there :s

**Review by Danarca — APPROVED** [review]
